### ISSUE TITLE
Wrong number of classes

### ISSUE THREAD

**Issue by rimio** [sirius-ai/MobileFaceNet_TF#24]
Please amend the source code or README file. The default number of classes (85164) is lower that the default dataset's actual number of classes (MS1M-refine-v2, 85742), which leads to failure to train.

I've basically re-implemented the model to reach this conclusion. After randomly getting NaN losses from softmax, tracing the issue to the loss function and for a few days investigating numerical stability for large number of classes, I found out that I was using the wrong shape for my logits.

This is an obscure issue, especially since the two numbers are very close. Please fix :).

**Comment by RichieMay** [comment]
the number of MS1M-refine-V1 classes is 85164.  there are also som problem when i use ms1m-v1.  can share your training pipeline?

**Comment by sirius-ai** [comment]
had fix：[](https://github.com/sirius-ai/MobileFaceNet_TF/commit/47a9855ef861407ae75fa171597e7636ae9a9285)
